test/js/bun/util/filesink.test.ts

```diff
@@ -146,3 +146,31 @@ describe("FileSink", () => {
     });
   }
 });
+
+import fs from "node:fs";
+import util from "node:util";
+import path from "node:path";
+
+it("end doesn't close when backed by a file descriptor", async () => {
```

Comment:
Where does it check the opposite - that Bun.file(path).writer().end() does close the file descriptor?

Reply:
this fails on posix as well. will look into it now for this branch, will look into fixing this for all platforms now
```ts
it("end does close when not backed by a file descriptor", async () => {
  const x = tmpdirSync();
  const file = Bun.file(path.join(x, "test.txt"));
  const writer = file.writer();
  await writer.write(Buffer.from("1 Hello, world!"));
  await writer.end();
  expect(async () => await writer.write(Buffer.from("2 Hello, world!"))).toThrow();
});
```

Reply:
okay, using `getMaxFD()` instead and `expect(fd_after).toBe(fd_before);` does pass on posix

Reply:
and does not on windows, i'll use that check instead